pkg/sqlite/repository.go

```diff
@@ -97,6 +98,10 @@ func (r *repository) runIdsQuery(ctx context.Context, query string, args []inter
 	for i, v := range result {
 		vsm[i] = v.Int
 	}
+
+	// We removed distinctIDs for postgresql, but now we have duplicates
+	vsm = sliceutil.AppendUniques(nil, vsm)
```

Comment:
Using sliceutil.AppendUnique is quite inefficient from a time complexity (performs a full scan of the slice for each element to add). Since this method is in a hot path, consider using a map to make this a lot faster:

```go
vsm := make(map[int]struct{}, len(result))
for _, v := range result {
    vsm[v.Int] = struct{}
}

return maps.Keys(vsm)
```

Reply:
The key is that it must retain the order of elements here, so i dont think a map is sufficient.

Reply:
Then how about this?

```go
found := make(map[int]struct{}, len(result))
vsm := make([]int, 0, len(result))
for _, v := range result {
    _, ok := found[v.Int]
    if !ok {
        found[v.Int] = struct{}
        vsm = append(vsm, v.Int)
    }
}

return vsm
```

Reply:
I did some benchmarks and for small number of results, this seems to actually be more efficient than using a lookup map. Ignoring for now